packages/discovery-react-components/src/components/StructuredQuery/__tests__/StructuredQuery.test.tsx

```diff
@@ -388,4 +448,58 @@ describe('<StructuredQuery />', () => {
       });
     });
   });
+
+  describe('On mount', () => {
+    it.only('Calls list fields', () => {
+      const mockListFields = jest.fn();
+      ({ structuredQuery } = setup({ apiOverrides: { listFields: mockListFields } }));
+      expect(mockListFields).toBeCalledTimes(1);
+      expect(mockListFields).toBeCalledWith();
+    });
+  });
+
+  describe('when there are project fields available', () => {
+    beforeEach(() => {
+      projectFields = [{ field: 'field name 1' }, { field: 'field name 2' }];
+    });
+
+    describe('and I choose field name 1 in the fieldname combobox', () => {
+      beforeEach(() => {
+        ({ structuredQuery } = setup({ projectFields }));
+        const input = structuredQuery.getByPlaceholderText('Select field');
+        fireEvent.change(input, { target: { value: 'field name 1' } });
+      });
+
+      it('should display field name 1', () => {
+        expect(structuredQuery!.getByText('field name 1')).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('when there are no project fields available', () => {
+    beforeEach(() => {
+      projectFields = undefined;
+    });
+
+    it('should display the loading text', () => {
+      ({ structuredQuery } = setup({ projectFields }));
+      expect(
+        structuredQuery.getByPlaceholderText(defaultMessages.fieldDropdownLoadingText)
+      ).toBeInTheDocument();
+    });
```

Comment:
Should we also check that it's disabled at this point?

Reply:
added